darwin/DarwinProcess.c

```diff
@@ -240,23 +241,28 @@ void DarwinProcess_setFromKInfoProc(Process* proc, const struct kinfo_proc* ps,
    proc->updated = true;
 }
 
+static double ticksToNanoseconds(const double ticks) {
+   const double nanos_per_sec = 1e9;
+   return ticks / (double) Platform_clockTicksPerSec * Platform_timebaseToNS * nanos_per_sec;
```

Comment:
`Platform_clockTicksPerSec * Platform_timebaseToNS * nanos_per_sec` is a runtime constant.
Maybe pre-compute it

Reply:
You mean something like?
```c
   static const double tickToNanosecondsScale = (double) Platform_clockTicksPerSec * Platform_timebaseToNS * 1e9;
   return ticks / tickToNanosecondsScale;
```

Reply:
I think this would not work, as `static` variables are initialized before `main` (and `Platform_clockTicksPerSec` and `Platform_timebaseToNS` are not yet set at this point).

Maybe it's OK to compute once every cycle when caching `time_interval` (see below).

Reply:
k, had to recheck on the actual initialization semantics here. Would take a second (`bool`) variable to act as lazy initialization guard to calculate this constant on first entry.

But as both values are set in `Platform_init` we can just initialize that scaling factor there.